File: app/src/main/java/com/task/paginationlist/data/network/NetworkNetworkSource.java
package com.task.paginationlist.data.network;

import com.task.paginationlist.data.network.netmodels.WallpaperList;

import io.reactivex.Observable;

public class NetworkNetworkSource {

    private NetworkApi api;

    public Observable<WallpaperList> getWallpaperList(int pageNum) {
        return api.getWallpaperList(pageNum);
    }

    public NetworkNetworkSource(NetworkApi api) {
        this.api = api;
    }
}

File: app/src/main/java/com/task/paginationlist/data/network/NetworkNetworkSource.kt
package com.task.paginationlist.data.network

import com.task.paginationlist.data.network.netmodels.WallpaperList

import io.reactivex.Observable

class NetworkNetworkSource(private val api: NetworkApi) {

    fun getWallpaperList(pageNum: Int): Observable<WallpaperList> {
        return api.getWallpaperList(pageNum)
    }
}
